cheatsheets/Password_Storage_Cheat_Sheet.md

```diff
@@ -1,567 +1,153 @@
+There are four main algorithms that should be considered for hashing passwords in modern applications. All of these algorithms support an iteration count of [work factor](#work-factors), which should be adjusted based on the system they are being used with.
 
-Java bindings by [phxql](https://github.com/phxql/argon2-jvm) has been used because it's currently the only binding proposed for Java in the [bindings list](https://github.com/P-H-C/phc-winner-argon2#bindings) and is simple to use.
+- [Argon2](https://github.com/P-H-C/phc-winner-argon2) is the winner of the 2015 [Password Hashing Competition](https://password-hashing.net), and should be used as the first choice when it is available. It has strong resistance to both GPU and ASIC based attacks.
+- [Scrypt](https://en.wikipedia.org/wiki/Scrypt) is designed to resist GPU based attacks, but is less widely supported.
+- [PBKDF2](https://en.wikipedia.org/wiki/PBKDF2) is recommended by [NIST](https://pages.nist.gov/800-63-3/sp800-63b.html#memsecretver), and should be used when FIPS compliance is required. It requires that 
+- [Bcrypt](https://en.wikipedia.org/wiki/Bcrypt) is the oldest of the algorithms, and is more susceptible to GPU based attacks. However, due to its age it is widely supported across most languages.
 
-[libsodium](https://github.com/jedisct1/libsodium) propose an implementation but it have been decided to use a dedicated project because libsodium provide several crypto features and thus work from maintainer will not focus on Argon2 implementation (however project is active and propose also bindings for many technologies).
+It should be stressed that even though Bcrypt is considered comparatively weak compared to newer algorithms such as Scrypt or Argon2, it is still substantially stronger than legacy algorithms such as MD5 and SHA-1. Although exact cracking speeds will vary based on the hardware, to give an idea of context, a benchmark using [8 Nvidia GTX 1080 GPUs](https://gist.github.com/epixoip/a83d38f412b4737e99bbef804a270c40) showed Bcrypt hashes to be approximately 2 million times harder to crack than MD5.
 
-## Remark about Argon2 native library embedding
+## Legacy Algorithms
 
-Due to the kind of data processed (password), the implementation without the embedded pre-compiled native library has been used in order to don't embed native untrusted compiled code (*there absolutely no issue with the project owner of argon2-jvm, it is just a secure approach*) that will be difficult to validate. For java part, sources are provided in Maven repositories along jar files.
+In some circumstances it is not possible to use [modern hashing algorithms](#modern-algorithms), usually due tot he use of legacy language or environments. Where possible, third party libraries should be used to provide these algorithms. However, if the only algorithms available are legacy ones such as MD5 and SHA-1, then there are a number of steps that can be taken to improve the security of stored passwords.
```

Comment:
I believe bcrypt is one of the legacy algorithms. It's been there for a good amount of time, and implemented across almost all platforms. Or even, `PBKDF2` implementation. Despite you adding a note to this at the end, I still can't see this actually happening. I can't not find a library that will implement any of the legacy slow KDFs

Reply:
Bcrypt is certainly older that some other (1999), but it's about the same age as AES (2001), which I wouldn't call legacy.

What I was aiming for here is the split between proper password hashing algorithms (work factors, salting, resistance to cracking) and things like MD5/SHA-1. Maybe legacy/modern isn't the best wording here - I was thinking more that the *approach* is legacy or modern (i.e, people used to recommend hashing passwords with MD5, but that's a very outdated recommendation now).

I'm not really sure better titles would be. `Modern Hashing Algorithms` vs `General Purpose Hashing Algorithms` (although they're arguably not) or `Fast Hashing Algorithms`?

While perhaps more technically correct, is this clearer to the reader?